### DISCUSSION TITLE
Multiple logons/connections for this session are not allowed. Closing connection from /XXXXX:11207 since session is already established from null

### DISCUSSION THREAD

**Discussion by kiropowered** [quickfix-j/quickfixj#748] [Q&A]
After a couple of days the fix engine stop working with the following message:

Multiple logons/connections for this session are not allowed. Closing connection from /XXXXX:11207 since session is already established from null

Then every 30sec(during the heartbeat I guess) it log the same error but with different port exemple : 
Multiple logons/connections for this session are not allowed. Closing connection from /XXXXX:11208 since session is already established from null
Multiple logons/connections for this session are not allowed. Closing connection from /XXXXX:11209 since session is already established from null
Multiple logons/connections for this session are not allowed. Closing connection from /XXXXX:11210 since session is already established from null

When restarting the fix engine the client is able to connect again

**Comment by chrjohn** [comment]
Which messages are exchanged before these log lines? I assume there is a heartbeat timeout? What was the last application message you received before this happened?

In general it looks like your session somehow broke and the counterparty tries to reconnect. Next time this happens a thread dump would be helpful.

**Comment by kiropowered** [comment]
We use quickfixj with jdbc(mysql), we shutdown the DB during night but not the app.
It seems like to happen when the db is restarted during the morning there is some logon messages with reset sequences and end up with the errors mentionned above.
So I assume it's due to that db shutdown, since it take quite some times to shutdown/restart the db it's pretty hard to assign periods in quickfixj.
Do you think shutting down the db can lead to this issue ?

**Comment by chrjohn** [comment]
For JDBC are you using Proxool (bundled with QFJ up to 2.3.1) or do you pass the data source to Spring Boot or the like?

I could imagine that it could have to do with the DB restart. What do you expect to happen when the data base is your session store and the DB is down?
Maybe you could adapt your session start/stop times to the DB restart?

**Comment by kiropowered** [comment]
Hello, I did try to reproduce the bug in dev mode by shuting down the db and restarting, trying to connect when db is off etc... But I can't get the same error.
Not sure how it happen and when the remote address can be null.
FYI we use the quickfixj-spring-boot-starter
